libcloud/common/base.py

```diff
@@ -76,6 +76,41 @@
 RETRY_FAILED_HTTP_REQUESTS = False
 
 
+class LazyObject(object):
+    """An object that doesn't get initialized until accessed."""
+
+    @classmethod
+    def _proxy(cls, *lazy_init_args, **lazy_init_kwargs):
+        class Proxy(cls, object):
+            _lazy_obj = None
+
+            def __init__(self):
+                # Must override the lazy_cls __init__
+                pass
+
+            def __getattribute__(self, attr):
+                lazy_obj = object.__getattribute__(self, '_get_lazy_obj')()
```

Comment:
Wrapping it in a method was nice, but I don't think you need to use ```__getattribute__``` to call that method since it's defined in the same class here, so any invocation of this method is guaranteed to have ```self._get_lazy_obj()``` available to call.